### ISSUE TITLE
Make `enable-new-git-file-globbing=true` the default and only option 

### ISSUE THREAD

**Issue by crenshaw-dev** [argoproj/argo-cd#14101]
# Summary

Make `enable-new-git-file-globbing=true` the only behavior. It's more consistent with our other globbing uses. https://github.com/argoproj/argo-cd/pull/13314/files

# Motivation

It's more intuitive. Using just the new globbing format will also eliminate one use of the `git` CLI, moving us further towards being able to go distroless.

# Proposal

Delete the code that did it the old way, leave the code that does it the new way.

**Comment by Hronom** [comment]
Agree, with current default option, I find my self easy to shoot in leg when you need to use sub-directories. So I need back and forth with google until not find [this](https://argo-cd.readthedocs.io/en/stable/operator-manual/applicationset/Generators-Git-File-Globbing/)

**Comment by rumstead** [comment]
I agree the globbing isn't intuitive though, I am fearful that this is asking for the appset controller to delete a bunch of applications without the operator's knowledge.

**Comment by crenshaw-dev** [comment]
@rumstead agreed. otoh I don't think letting both modes live forever is a good option. I think we should have release notes that 1) make it super easy to determine if you have any appsets which may be impacted and 2) explains how to check the before/after behavior to avoid accidental deletions.

**Comment by rumstead** [comment]
A possible bug when using globbing - #14101

**Comment by reggie-k** [comment]
As per discussion with @crenshaw-dev, further thought is needed on a safe migration path. Removing from 3.0.
